cptvtool/main.go

```diff
@@ -21,37 +35,39 @@ func runMain() error {
 	if len(os.Args) != 2 {
 		return fmt.Errorf("usage: %s <filename>", os.Args[0])
 	}
-	file, err := os.Open(os.Args[1])
-	if err != nil {
-		return err
-	}
-	defer file.Close()
-	bfile := bufio.NewReader(file)
-	r, err := cptv.NewParser(bfile)
-	if err != nil {
-		return err
-	}
-	fields, err := r.Header()
+
+	fr, err := cptv.NewFileReader(os.Args[1])
 	if err != nil {
 		return err
 	}
+	defer fr.Close()
 
-	fmt.Println(fields.Timestamp(cptv.Timestamp))
-	fmt.Println(fields.Uint32(cptv.XResolution))
-	fmt.Println(fields.Uint32(cptv.YResolution))
-	fmt.Println(fields.Uint8(cptv.Compression))
-	fmt.Println(fields.String(cptv.DeviceName))
+	fmt.Println("Timestamp:   ", fr.Timestamp())
+	fmt.Println("Device Name: ", fr.DeviceName())
 
+	// Read the frames and get a frame count. This is an illustration of
+	// frame reading - the r.FrameCount method will do the same thing (and
+	// will similarly leave the file pointer at EOF)
 	frames := 0
+	var frame lepton3.Frame
 	for {
-		_, _, err := r.Frame()
-		if err == io.EOF {
-			break
-		} else if err != nil {
+		err := fr.ReadFrame(&frame)
+		if err != nil {
+			if err == io.EOF {
+				fmt.Print("<EOF>")
+				frames++ // the last valid read returns EOF
+				break
+			}
 			return err
 		}
 		frames++
+		fmt.Print(".")
+		if frames%80 == 0 {
+			fmt.Print("\n")
+		}
```

Comment:
There's probably no need for the newlines. Not everyone is going to have 80 char width terminals so just let the terminal handle the wrapping.